app/app/src/main/java/de/stustapay/stustapay/net/TerminalApiAccessorInner.kt

```diff
@@ -0,0 +1,136 @@
+import de.stustapay.api.apis.CashierApi
+import de.stustapay.api.apis.CustomerApi
+import de.stustapay.api.apis.OrderApi
+import de.stustapay.api.apis.UserApi
+import de.stustapay.stustapay.model.RegistrationState
+import de.stustapay.stustapay.storage.RegistrationLocalDataSource
+import io.ktor.client.HttpClientConfig
+import io.ktor.client.engine.cio.CIO
+import io.ktor.client.plugins.HttpRequestRetry
+import io.ktor.client.plugins.HttpTimeout
+import io.ktor.client.plugins.contentnegotiation.ContentNegotiation
+import io.ktor.client.plugins.logging.LogLevel
+import io.ktor.client.plugins.logging.Logger
+import io.ktor.client.plugins.logging.Logging
+import io.ktor.serialization.kotlinx.json.json
+import kotlinx.coroutines.flow.first
+import kotlinx.serialization.json.Json
+
+class TerminalApiAccessorInner(
+    private val registrationLocalDataSource: RegistrationLocalDataSource,
+    private val retry: Boolean = false,
+    private val logRequests: Boolean = true
+) {
+    private var authApi =
+        AuthApi("http://localhost", CIO.create { this.https {} }) { configureApi(it) }
+    private var baseApi =
+        BaseApi("http://localhost", CIO.create { this.https {} }) { configureApi(it) }
+    private var cashierApi =
+        CashierApi("http://localhost", CIO.create { this.https {} }) { configureApi(it) }
+    private var customerApi =
+        CustomerApi("http://localhost", CIO.create { this.https {} }) { configureApi(it) }
+    private var orderApi =
+        OrderApi("http://localhost", CIO.create { this.https {} }) { configureApi(it) }
+    private var userApi =
+        UserApi("http://localhost", CIO.create { this.https {} }) { configureApi(it) }
+
+    fun configureApi(conf: HttpClientConfig<*>) {
+        conf.install(ContentNegotiation) {
+            json(Json {
+                prettyPrint = true
+                isLenient = true
+                ignoreUnknownKeys = true
+            })
+        }
+
+        if (retry) {
+            conf.install(HttpRequestRetry) {
+                retryOnServerErrors(maxRetries = 5)
+                retryOnException(maxRetries = 3, retryOnTimeout = true)
+                exponentialDelay()
+            }
+        }
+
+        conf.install(HttpTimeout) {
+            connectTimeoutMillis = 5000
+            requestTimeoutMillis = 10000
+            socketTimeoutMillis = 10000
+        }
+
+        if (logRequests) {
+            conf.install(Logging) {
+                level = LogLevel.ALL
+                logger = object : Logger {
+                    override fun log(message: String) {
+                        Log.d("StuStaPay req", message)
+                    }
+                }
+            }
+        }
+
+        conf.expectSuccess = false
+
+        conf.install(Logging)
+    }
+
+    suspend fun checkRegistration(): Boolean {
+        return when (val reg = registrationLocalDataSource.registrationState.first()) {
+            is RegistrationState.Registered -> {
```

Comment:
registering all the apis and access tokens again for every reference to `.auth()` or `.cashier()` seems wrong - these instances should be reused, otherwise we create new api instances for every request.

Reply:
Done